### PR TITLE
Feature/addingbitcorefirstsanitychecks

### PR DESCRIPTION
**Pull request by Levino**
Adding local checks for right format on bitcoin addresses and so on as to reduce the amount of "bad responses" and difficulties on error handling.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,9 +21,9 @@
   },
   "homepage": "https://github.com/chain-engineering/chain-node",
   "dependencies": {
-    "request": "2.36.0",
-    "bitcoinjs-lib": "1.2.0",
```
Comment (ryandotsmith) [inline package.json]: I'mt not entirely convinced that we should move away from this lib. What is the motivation? Also, we depend on this lib in the signer.js file.

File: index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-var bitcoin = require('bitcoinjs-lib');
 var HttpUtility = require('./http-utility');
 var Signer = require('./signer');
-var util = require('util');
+var util = require('util'),
```
Comment (ryandotsmith) [inline index.js]: I'd like to stick to one style for require statements. So far we have 1 require / `;` per line. We should pick one and go with it. Unless there is a strong argument for using `,` terminated lines - I'd like to stick to `;` terminated lines.

**Comment by ryandotsmith** [comment]
@Levino Thanks for the PR. I've made a few inline comments. In general, I am in favor of adding validation checks prior to making network requests - we can certainly save some time and bandwidth by doing this... However, I might be inclined to go with a slightly different implementation. Adding this feature might be as simple as a 3 line diff if we use the existing tooling.

File: package.json (hunk 1 above)
Comment (Levino) [inline package.json]: Sorry. Didnt realize the other dependency. Hooray for automatic tesing :) I can just add them again. I really like bitcore. Cant really say why.

File: index.js (hunk 2 above)
Comment (Levino) [inline index.js]: Okay. Just a provocation on my part :)

**Comment by Levino** [comment]
I would remove the example.js if I had test. Could you merge #17 ?

**Comment by ryandotsmith** [comment]
We'll close this for now. We'd like to have some sanity checks in place before doing network requests, but we'd like not to have to add the async and bitcore dependencies.

**Pull request closed without merging**